packages/composer/amazeelabs/silverback_gutenberg/src/Plugin/GraphQL/DataProducer/EditorBlockMedia.php

```diff
@@ -26,31 +33,122 @@
+    );
+  }
+
+  /**
+   * EntityLoad constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration array.
+   * @param string $pluginId
+   *   The plugin id.
+   * @param array $pluginDefinition
+   *   The plugin definition array.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityRepositoryInterface $entityRepository
+   *   The entity repository service.
+   * @param \Drupal\graphql\GraphQL\Buffers\EntityBuffer $entityBuffer
+   *   The entity buffer service.
+   *
+   * @codeCoverageIgnore
+   */
+  public function __construct(
+    array $configuration,
+          $pluginId,
+    array $pluginDefinition,
+    EntityTypeManagerInterface $entityTypeManager,
+    EntityRepositoryInterface $entityRepository,
+    EntityBuffer $entityBuffer
   ) {
-    $mediaId = $block['attrs']['mediaEntityIds'][0] ?? NULL;
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->entityTypeManager = $entityTypeManager;
+    $this->entityRepository = $entityRepository;
+    $this->entityBuffer = $entityBuffer;
+  }
 
-    if (!$mediaId) {
-      return NULL;
-    }
-    $media = Media::load($mediaId);
-    // Try to get the embedded item in the current language or use whatever
-    // we get as a fallback.
-    if (!$media) {
-      \Drupal::logger('graphql_gutenberg')->notice("Cannot load media by ID '{$mediaId}'");
+  public function resolve($block, FieldContext $context): ?Deferred {
+    $id = $block['attrs']['mediaEntityIds'][0] ?? NULL;
+
+    if (!$id) {
       return NULL;
     }
-    $documentLanguage = $field->getContextValue('document_language');
-    if ($media->hasTranslation($documentLanguage)) {
-      $media = $media->getTranslation($documentLanguage);
-    }
-    if ($media) {
-      $field->addCacheableDependency($media);
-    }
-    return $media;
+
+    $resolver = $this->entityBuffer->add('media', $id);
+    $language = $context->getContextValue('document_language');
+
+    return new Deferred(function () use ($language, $resolver, $context) {
+      if (!$entity = $resolver()) {
+        // If there is no entity with this id, add the list cache tags so that
+        // the cache entry is purged whenever a new entity of this type is
+        // saved.
+        $type = $this->entityTypeManager->getDefinition('media');
+        /** @var \Drupal\Core\Entity\EntityTypeInterface $type */
+        $tags = $type->getListCacheTags();
+        $context->addCacheTags($tags);
+        return NULL;
+      }
+
+      $context->addCacheableDependency($entity);
+      if (isset($bundles) && !in_array($entity->bundle(), $bundles)) {
+        // If the entity is not among the allowed bundles, don't return it.
+        return NULL;
+      }
+
+      // Get the correct translation.
+      if (isset($language) && $language !== $entity->language()->getId() && $entity instanceof TranslatableInterface) {
```

Comment:
`$entity->hasTranslation($language)` check is missing. `getTranslation` can throw without the check.